app/demo-loop/Runner.cc

```diff
@@ -77,17 +77,19 @@ Runner::Runner(RunnerInput const& inp,
 
     ScopedRootErrorHandler scoped_root_error;
     this->build_core_params(inp, std::move(output));
-    this->build_step_collectors(inp);
-    this->build_transporter_input(inp);
-    this->build_primaries(inp);
-    use_device_ = inp.use_device;
-
-    if (root_manager_)
+    if (!inp.mctruth_filename.empty())
     {
-        // Store input and CoreParams data
+        // Initialize ROOT file and step collector
+        root_manager_
+            = std::make_shared<RootFileManager>(inp.mctruth_filename.c_str());
+        this->build_step_collectors(inp);
```

Comment:
No. We need to call this even if ROOT isn't active: there are other step collectors. I don't understand why this code was changed here and below?